### PR TITLE
Split and rename generated "factories" and "provider"

### PR DESCRIPTION
**Pull request by JakeWharton**
In order to support global modifiers, we need make changes to these types in a way that does not work with their current setup. In anticipation of that, rename them to give distinct behavior.

🤞 this is the last change pulled out/ahead of #1084

---

- [x] `CHANGELOG.md`'s "Unreleased" section has been updated, if applicable.

### CODE REVIEW

File: redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/protocolGuestGeneration.kt [hunk 1]
```diff
@@ -47,11 +47,11 @@ import com.squareup.kotlinpoet.UNIT
 import com.squareup.kotlinpoet.joinToCode
 
 /*
-class ExampleProtocolBridge(
+class ExampleProtocolBridge private constructor(
```
Comment (JakeWharton) [inline redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/protocolGuestGeneration.kt:50]: Drive-by cleanup

File: redwood-compose/src/commonMain/kotlin/app/cash/redwood/compose/RedwoodComposition.kt [hunk 2]
```diff
@@ -58,7 +60,7 @@ public interface RedwoodComposition {
 public fun <W : Any> RedwoodComposition(
   scope: CoroutineScope,
   view: RedwoodView<W>,
-  provider: Widget.Provider<W>,
+  widgetSystem: WidgetSystem<W>,
```
Comment (swankjesse) [inline redwood-compose/src/commonMain/kotlin/app/cash/redwood/compose/RedwoodComposition.kt:63]: YAASS

**Review by swankjesse — APPROVED** [review]

**Review by colinrtwhite — APPROVED** [review]
